src/iterated_integrals.jl

```diff
@@ -0,0 +1,345 @@
+abstract type AbstractWikJ end
+abstract type AbstractWikJDiagonal <: AbstractWikJ end
+abstract type AbstractWikJCommute <: AbstractWikJ end
+abstract type AbstractWikJGeneral <: AbstractWikJ end
+
+struct WikJDiagonal_oop <: AbstractWikJDiagonal end
+
+mutable struct WikJDiagonal_iip{WikJType} <: AbstractWikJDiagonal
+    WikJ::WikJType
+    WikJDiagonal_iip(ΔW) = new{typeof(ΔW)}(false .* ΔW .* ΔW)
+end
+
+struct WikJCommute_oop <: AbstractWikJCommute end
+
+mutable struct WikJCommute_iip{WikJType} <: AbstractWikJCommute
+    WikJ::WikJType
+    function WikJCommute_iip(ΔW)
+        WikJ = false .* ΔW .* ΔW'
+        new{typeof(WikJ)}(WikJ)
+    end
+end
+
+struct WikJGeneral_oop <: AbstractWikJGeneral
+    m_seq::Matrix{Int}
+    function WikJGeneral_oop(ΔW)
+        m = length(ΔW)
+        M = Int(m*(m-1)/2)
+        m_seq = Matrix{Int}(undef, M, 2)
+        k = 1
+        for i in 1:length(ΔW)
+          for j in i+1:length(ΔW)
+            m_seq[k,1] = i
+            m_seq[k,2] = j
+            k += 1
+          end
+        end
+        new(m_seq)
+    end
+end
+
+mutable struct WikJGeneral_iip{rateNoiseElTypeNoUnits, WikJType} <: AbstractWikJGeneral
+    WikJ::WikJType
+    WikJ2::WikJType
+    WikJ3::WikJType
+    m_seq::Matrix{Int}
+    vec_ζ::Vector{rateNoiseElTypeNoUnits}
+    vec_η::Vector{rateNoiseElTypeNoUnits}
+    Gp₁::Vector{rateNoiseElTypeNoUnits}
+    Gp₂::Vector{rateNoiseElTypeNoUnits}
+    Aᵢ::Vector{rateNoiseElTypeNoUnits}
+end
+
+function WikJGeneral_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJ2 = false .* ΔW .* ΔW'
+    WikJ3 = false .* ΔW .* ΔW'
+    m = length(ΔW)
+    M = Int(m*(m-1)/2)
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    vec_ζ = false .* vec(ΔW)
+    vec_η = false .* vec(ΔW)
+    Gp₁ = false .* Array{eltype(ΔW)}(undef, M)
+    Gp₂ = false .* Array{eltype(ΔW)}(undef, M)
+    Aᵢ = false .* vec(ΔW)
```

Comment:
Those `false .*` are unnecessary and inefficient, because the desired array has the same shape with `ΔW`, so `zero` and `fill! + similar` is sufficient.